internals/link_helpers.py

```diff
@@ -0,0 +1,86 @@
+import re
+import requests
+import json
+import logging
+from typing import Any
+
+LINK_TYPE_CHROMIUM_BUG = 'chromium_bug'
+LINK_TYPE_UNKNOWN = 'unknown'
+LINK_TYPES_REGEX = {
+    # https://bugs.chromium.org/p/chromium/issues/detail?id=
+    LINK_TYPE_CHROMIUM_BUG: re.compile(r'https://bugs\.chromium\.org/p/chromium/issues/detail\?.*'),
+    # any other links
+    LINK_TYPE_UNKNOWN: re.compile(r'https?://.*'),
+}
+
+
+class Link():
+
+  @classmethod
+  def get_type(cls, link: str) -> str | None:
+    """Return Link_Type if the given link is valid. Otherwise, return None."""
+    for link_type, regex in LINK_TYPES_REGEX.items():
+      if regex.match(link):
+        return link_type
+    return None
+
+  def __init__(self, url: str):
+    self.url = url
+    self.type = Link.get_type(url)
+    self.is_parsed = False
+    self.is_error = False
+    self.information = None
+
+  def _parse_chromium_bug(self) -> dict[str, object]:
+    """Parse the information from the chromium bug tracker."""
+
+    endpoint = 'https://bugs.chromium.org/prpc/monorail.Issues/GetIssue'
+
+    issue_id = self.url.split('id=')[-1]
+
+    # csrf token is required, its expiration is about 2 hours according to the tokenExpiresSec field
+    # technically, we could cache the csrf token and reuse it for 2 hours
+
+    csrf_token = re.findall(
```

Comment:
I think this would only work for a request from a user that is currently signed in to bugs.chromium.org.   When our python server makes such a request, there is no signed in user, so instead it will generate a redirect to a login page. 

 You could try the same trick with a request to `"https://bugs.chromium.org/"`.

It would be better for us to register as a monorail API client using oauth.  But we can leave that as a TODO comment for now.

Reply:
@jrobbins 

It will get valid `csrf_token` even it is not signed in, `requests.get` will not follow redirection by default.

Prove:
- ![CleanShot 2023-05-29 at 12 16 29](https://github.com/GoogleChrome/chromium-dashboard/assets/5123601/2e86ecfc-5034-44d8-b47f-927b5304ed6f)

- If token is not valid, `test_parse_chromium_tracker` won't pass.

I notices some issues requires permission to view.

We can leave a TODO to use an API client.